state_machine.py

```diff
@@ -0,0 +1,81 @@
+import datetime
```

Comment:
everything looks good in this file; 
fun exercise for advanced python knowledge: checkout python decorators, and how you can use them to improve the functions in this file;

Reply:
marking this file as viewed - if you do implement decorators, I'll review again;